Rating/mec_bpc.py

```diff
@@ -0,0 +1,123 @@
+import json
+from time import sleep
+from tqdm import tqdm
+import openai
+from string import Template
+from typing import Optional, List, Union
+
+
+MAX_API_RETRY = 5
+REQ_TIME_GAP = 5
+
+
+def clean_continuation(rating):
+    rating = rating.split('/')[0].split('out')[0].strip()
+    if rating[-1] == '.':
+        rating = rating[:-1]
+    return rating
+
+
+def extract_rating(rating):
+    try:
+        assist1 = float(
+            clean_continuation(rating.split("Assistant 1:")[1].split("\n")[0]))
+        assist2 = float(
+            clean_continuation(rating.split("Assistant 2:")[1].split("\n")[0]))
+        return assist1-assist2
+    except ValueError as err:
+        print(err)
+        return None
+    except IndexError as err:
+        print(err)
+        return None
+
+
+def get_eval(system_prompt, user_prompt: str):
+    ratings = [None for _ in range(3)]
+    for rating_idx in range(3):
+        retry_count = 0
+        for _ in range(MAX_API_RETRY):
+            try:
+                messages = [{"role": "system", "content": system_prompt}] if system_prompt else []
+                messages.append({"role": "user", "content": user_prompt})
+                response = openai.ChatCompletion.create(
+                    model="gpt-3.5-turbo",
+                    messages=messages,
+                    temperature=0.0,
```

Comment:
I think `temperature` should be positive for the three repeated generations, however I'm seeing that for this model they are still going to be distinct, but only slightly:
```
"critique": [
    "Evaluation evidence: \n\nBoth Assistant 1 and Assistant 2 provided correct and relevant answers to the question. Assistant 1 asked for clarification on the language preference, which shows attentiveness to the user's needs. Assistant 2 directly provided the correct answer without any additional information.\n\nThe helpfulness of both responses is high as they directly answer the user's question. The relevance is also high as the capital of France is a key piece of information related to the question. The accuracy is perfect for both assistants as they both correctly identified Paris as the capital of France. However, in terms of level of detail, Assistant 2's response is lacking as it only provides the capital city without any additional information.\n\nBased on these factors, I would rate Assistant 1 with a score of 9 and Assistant 2 with a score of 8.\n\nThe score of Assistant 1: 9\nThe score of Assistant 2: 8",
    "Evaluation evidence: \n\nBoth Assistant 1 and Assistant 2 provided correct and relevant answers to the question. Assistant 1 asked for clarification on the language preference, which shows attentiveness to the user's needs. Assistant 2 directly provided the correct answer without any additional information.\n\nThe helpfulness of both responses is high as they directly answer the user's question. The relevance is also high as the capital of France is a key piece of information related to the question. The accuracy is perfect for both assistants as they both correctly identified Paris as the capital of France. However, in terms of level of detail, Assistant 2's response is lacking as it only provides the capital city without any additional information.\n\nBased on these factors, I would rate Assistant 1 and Assistant 2 as follows:\n\nThe score of Assistant 1: 9\nThe score of Assistant 2: 8",
    "Evaluation evidence: \n\nBoth Assistant 1 and Assistant 2 provided correct and relevant answers to the question. Assistant 1 asked for clarification on the language preference, which shows attentiveness to the user's needs. Assistant 2 directly provided the correct answer without any additional information.\n\nThe score of Assistant 1: 8\nThe score of Assistant 2: 9"
],

```